src/services/notification.ts

```diff
@@ -16,55 +16,170 @@
+    public static $inject = ['$log', '$window', '$state', 'SettingsService'];
+
+    constructor($log: ng.ILogService, $window: ng.IWindowService,
+                $state: ng.ui.IStateService, settingsService: SettingsService) {
         this.$log = $log;
         this.$window = $window;
         this.$state = $state;
+        this.settingsService = settingsService;
     }
 
     /**
      * Ask user for desktop notification permissions.
      *
      * Updates internal `maxNotify` flag. If the user accepts, the is set to to
      * true. If the user declines, the flag is set to false. If the
-     * Notifications API is not available, the flag is set to null.
+     * user makes no choice, the flag is set to null
      */
-    public requestNotificationPermission(): void {
-        if (!('Notification' in this.$window)) {
-            // API not available
-            this.mayNotify = null;
-        } else {
-            const Notification = this.$window.Notification;
-            if (Notification.permission === 'granted') {
-                // Already granted
-                this.mayNotify = true;
-            } else if (Notification.permission === 'denied') {
-                // Not granted
-                this.mayNotify = false;
+    private requestNotificationPermission(): void {
+        const Notification = this.$window.Notification;
+        this.$log.debug(this.logTag, 'Requesting notification permission...');
+        Notification.requestPermission((result) => {
+            switch (result) {
+                case 'denied':
+                    this.notificationPermission = false;
+                    break;
+                case 'granted':
+                    this.notificationPermission = true;
+                    this.desktopNotifications = true;
+                    this.storeSetting(NotificationService.SETTINGS_NOTIFICATIONS, 'true');
+                    break;
+                case 'default':
+                    this.desktopNotifications = false;
+                    this.notificationPermission = null;
+                    break;
+                default:
+                    this.notificationPermission = false;
+                    break;
+            }
+            this.$log.debug(this.logTag, 'Notification permission', this.notificationPermission);
+        });
+    }
+
+    /**
+     * Check the notification api availability and permission state
+     *
+     * Denied: Don´t ask again, user must manually re-grant permission
+     * via browser settings
+     * Granted: We have the (browser-)permission to send notifications
+     * Default: Initially state on first visit
+     */
+    public checkNotificationAPI(): void {
+        this.notificationAPIAvailable = ('Notification' in this.$window);
+        this.$log.debug(this.logTag, 'Notification API available:', this.notificationAPIAvailable);
+        const Notification = this.$window.Notification;
+        switch (Notification.permission) {
```

Comment:
You'll get a TypeError here if the notification API is not available.

In theory we don't support browsers without a desktop notification system, but there may also be strange configurations around, or people with funny plugins blocking stuff.